addons/mail/static/src/messaging/entity/composer/composer.js

```diff
@@ -50,6 +84,27 @@ function ComposerFactory({ Entity }) {
             });
         }
 
+        /**
+         * @param mentionKeyword
+         */
+        insertSuggestedPartner(mentionKeyword) {
+            const selection = this.suggestedPartners.find(suggestedPartner => suggestedPartner == this.activePartner);
+            const cursorPosition = this.textInputCursorStart;
+            const textLeft = this.textInputContent.substring(0, cursorPosition - (mentionKeyword.length));
+            const textRight = this.textInputContent.substring(cursorPosition, this.textInputContent.length);
+            const insert = selection.name.replace(/ /g, '\u00a0');
+            this.update({
+                suggestedPartners: [['unlink-all']],
+                textInputContent: textLeft + insert + ' ' + textRight,
+                textInputCursorStart: textLeft.length + selection.name.length + 2,
+                textInputCursorEnd: textLeft.length + selection.name.length + 2,
+            });
+            const mention = this.mentionedPartners.find(item => item.id === selection.id);
```

Comment:
You can compare entities directly, no need to compare ids.
Also I would rename `item` to a relevant name such as `mentionedPartner`, similarly I would rename the returned value to `mentionedPartner` instead of `mention`.